### ISSUE TITLE
Firewall with oauth2 enabled doesn't enforce valid authentication token

### ISSUE THREAD

**Issue by rene-roesch-blg** [thephpleague/oauth2-server-bundle#114]
Hi,

the currently suggested security configuration doesn't set up a firewall or access control in a way that enforces the oauth2 authentication token to be present on requests. Instead, it only checks its validity *if* present.

Based on the installation guide and basic setup my config looks like this (removed application specific parts not related to oauth2):
```
security:
    providers:
        app_user_provider:
            entity:
                class: App\Entity\Main\AppUser
                property: email
    firewalls:
        api_token:
            pattern: ^/api/v1/oauth2/token$
            security: false
        api_private:
            pattern: ^/api/v1/private
            security: true
            stateless: true
            oauth2: true
            provider: app_user_provider

    enable_authenticator_manager: true
```

Now, whenever I'm calling a route starting with `/api/v1/private` the `League\Bundle\OAuth2ServerBundle\Security\Authenticator\OAuth2Authenticator` is being called. As expected.
However, its `supports` methods returns `false` if not an invalid or expired, but no token at all was provided:

```
    public function supports(Request $request): ?bool
    {
        return 0 === strpos($request->headers->get('Authorization', ''), 'Bearer ');
    }
```

If I just don't send the `Authorization` header at all, all those endpoints are publicly accessible.
The only way for me to prevent this is to add an explicit `access_control` directive to let Symfony handle that:

```
    access_control:
        - { path: ^/api/v1/private, roles: IS_AUTHENTICATED_FULLY }
```

An example like this is being mentioned in the installation docs regarding the `authorize` endpoint specifically, but not generally for protecting all routes.

I'm wondering if the `OAuth2Authenticator` class is actually fine by *only* authenticating whenever the `Authorization` header is present or whether it should also consider cases where the header is simply missing.
Personally I was expecting it to immediately return 401/403 if the header is missing given how convenient it seems to be configurable adding `oauth2: true` to your firewall.

So, depending on how you deem the Authenticator's responsibilities, this may not be a bug.
If so, I'd at least want to offer a PR adding that piece of information to the docs to make sure people configure some sort of `access_control` themselves.

--
Edit:
This issue was pointed out by someone else already: https://github.com/thephpleague/oauth2-server-bundle/issues/94

The comments didn't however address the issue at hand but rather talked about the header's missing in PHP's context due to misconfigured webservers. I'm not talking about that.

**Comment by Amunak** [comment]
That's literally how it's supposed to work.

You can have multiple authenticators in a given firewall and you want to make sure that only the one applicable to the type of authentication the user is doing happens.

You could have, say, a HTTP Basic Auth authenticator on the same firewall (which also uses the Authorization header) and want to activate only the basic auth authenticator, not the OAuth one.

And when you have even a login form or whatever you still need to create rules with access control (or security checks like "isGranted") for it to actually ask for authentication.

In other words you are mixing up authentication and access control. They are two separate (if related) things.

**Comment by chalasr** [comment]
Closing as explained.
